### DISCUSSION TITLE
502 Bad Gateway nginx

### DISCUSSION THREAD

**Discussion by blogdemoi** [mcguirepr89/BirdNET-Pi#511] [Q&A]
Hi,

I started my Birdnet-Pi station two days ago and already had to hard restart (unplug/plug in from power) two times because I have this error message on my Birdnet-Pi domain web page:

> 502 Bad Gateway
> nginx

What can be wrong?

Thank you.

**Comment by mcguirepr89** [comment]
Could you please tell me where in your network setup the nginx server comes into play? I'm afraid nginx support is outside of the scope of what I'm able to help with for BirdNET-Pi

**Reply by blogdemoi** [reply]
> Could you please tell me where in your network setup the nginx server comes into play? 

Do you mean there is no nginx server on the Birdnet-Pi? Oh, I didn't read the doxumentation enough, you are using a Caddy server.

So, it must come from my main server that redirects my Birdnet-Pi station to the outside world. As I already self-host another server on a Pi and my Internet box can only accept one server behind it (Hope it is not too confusing :-)).

**Reply by mcguirepr89** [reply]
Yeah, the error was from your attempt to route traffic through nginx. I would recommend reading up on Caddy and allow it to sit in front as your new reverse_proxy as it is superior in all ways, especially with translating headers, which is likely the issue nginx is complaining about.

~~I'm going to close this issue since BirdNET-Pi doesn't have anything to do with nginx.~~
